Modix/Modules/CatModule.cs

```diff
@@ -0,0 +1,178 @@
+        {
+            var obj = new URL();
+            var json = string.Empty;
+            var flag = false;
+
+            // five chances to find a cat gif
+            for (var i = 0; i < 5; i++)
+            {
+                json = await DownloadCatJson(token);
+
+                // Check and make sure the string isn't empty before attempting to deserialize cat
+                if (string.IsNullOrWhiteSpace(json))
+                {
+                    Log.Error("Empty JSON String");
+                    return;
+                }
+
+                obj = DeserializeJson(json);
+
+                if (obj.file.EndsWith(".gif"))
+                {
+                    flag = true;
+                    break;
+                }
+            }
+
+            if (flag)
+            {
+                await Context.Channel.SendMessageAsync(obj.file);
+            }
+            else
+            {
+                await Context.Channel.SendMessageAsync("404 cat gif not found");
+            }
+        }
+
+        private async Task GetCatPicture(CancellationToken token)
+        {
+            var json = string.Empty;
+            var obj = new URL();
+            var flag = false;
+
+            // 5 chances to find a cat picture
+            for (var i = 0; i < 5; i++)
+            {
+                json = await DownloadCatJson(token);
+
+                // Check and make sure the string isn't empty before attempting to deserialize cat
+                if (string.IsNullOrWhiteSpace(json))
+                {
+                    Log.Error("Empty JSON String");
+                    return;
+                }
+
+                obj = DeserializeJson(json);
+
+                if (!obj.file.EndsWith(".gif"))
+                {
+                    flag = true;
+                    break;
+                }
+            }
+
+            if (flag)
+            {
+                await DownloadCatPicture(obj.file, token);
+            }
+            else
+            {
+                await Context.Channel.SendMessageAsync("404 cat picture not found");
+            }
+        }
+
+        private async Task<string> DownloadCatJson(CancellationToken token)
+        {
+            var json = string.Empty;
+
+            try
+            {
+                using (var response = await Client.GetAsync("http://random.cat/meow", token))
+                {
+                    if (response.IsSuccessStatusCode)
+                    {
+                        json = await response.Content.ReadAsStringAsync();
+                    }
+                    else
+                    {
+                        // If there is a bad result, an empty json string will be returned
+                        Log.Error("Invalid HTTP Status Code");
+                    }
+                }
+            }
+            catch (HttpRequestException hre)
+            {
+                Log.Fatal($"Request Exception", hre.InnerException);
```

Comment:
Warning level is fine here, though you may want to respond to the channel with a sane error message.